### ISSUE TITLE
Missing form-data when multiple inputs of same name

### ISSUE THREAD

**Issue by wildunne** [svenkubiak/mangooio#256]
I'm using `Taggle.js` in order to enable users on a site to add tags
to some content in a moderately visual way. It automatically creates
inputs which all share the same name.

Currently Mangoo.io is just providing me with one of the items - and 
they aren't seen when for example looping through the form params

I can of course work around this temporarily, but it would be nice if 
it would return a list or something like this

**Comment by svenkubiak** [comment]
Yes, you are right. Multiple values in Forms are not handled quiet correctly. Simple fix would be to just add a value list and retrieve it. But I am thinking on how to handle validations correctly with multiple values.

**Comment by wildunne** [comment]
Can't think of any particularly good ways of handling it unfortunately

**Comment by svenkubiak** [comment]
Ok, this is defnitily something we need to address in mangoo I/O, as forms with arrays (e.g. foo[]) are not working. Unfurtnatly this does currently not work with validation.

**Comment by wildunne** [comment]
What exactly do you mean by validation? I'll happily try and have a 
crack at it if I understand a little better, as it's a feature I asked for

**Comment by svenkubiak** [comment]
I mean the form field validation, like

"field foo can not be blank"

Currently, the validation is based on **field** <-> **value** mapping. But if we have multiple values, the validation does not work at the moment. For example in a scenario like:

```
<form action="/bar" method="post">
<input type="checkbox" name="foo[]" value="a">
<input type="checkbox" name="foo[]" value="b">
<input type="checkbox" name="foo[]" value="c">
</form> 
```

**Comment by wildunne** [comment]
I would guess that the best way to apply it would just to make 
sure that it's not empty? i.e if there is one non-empty entry in the 
list it's considered valid

**Comment by svenkubiak** [comment]
But from a usable view you could use all validation on the field, e.g. isNumber or isEmail, etc. This could make it confusing for a developer. I would suggest either have all possible validations or no-validation for multivalue fields.

**Comment by wildunne** [comment]
I suppose that would make sense. It's quite a niche use-case so I 
don't think it would be too much of a pain to force the developer
to implement their own validations on them.

**Comment by svenkubiak** [comment]
Agree. No validation for multivalue form fields.

**Comment by svenkubiak** [comment]
I am not sure, if undertow can handle multivalued forms. There is at least not test case for this:

https://github.com/undertow-io/undertow/blob/master/core/src/test/java/io/undertow/server/handlers/form/FormDataParserTestCase.java

**Comment by svenkubiak** [comment]
Ok. Finally figured it out. Undertow correctly supports it. It was wrongly parsed in the FormHandler.
